agent/ansible/pbench/agent/roles/pbench_firewall_open_ports/tasks/main.yml

```diff
@@ -0,0 +1,12 @@
+---
+# pbench agent configuration
+- name: "pbench agent configuration - open firewall ports for tool meister"
+  ansible.posix.firewalld:
```

Comment:
Is this sensitive to whether `firewalld` is installed or running, or is extra logic required? (Probably not an issue if we require someone to run this manually; but definitely something we should think about if it's run implicitly from the main install playbook.)

Reply:
This is not sensitive to whether the firewall is running: it will unconditionally modify the configuration (that's problem 1). It will *not* restart `firewalld` however (that's problem 2 - or not, depending on your POV), so the settings will *not* take effect until it is restarted. I could make the settings `immediate`, but then the playbook would fail if the firewall is not running.

So the current state is:
- the settings will be unconditionally applied, but will take effect if `firewalld` is (re)started.
- the role is invoked by the `pbench_agent_config` role automatically: that's for testing convenience, but it's not going to fly for actual use.

We could  put the onus on the user: if they are running `firewalld`, we ask them to run the (separate) playbook, which not only opens the ports, but also restarts `firewalld` (or opens the ports `immediate`ly). OTOH, if they are not running `firewalld`, they don't have to run the playbook at all.